packages/ld/ld.ts

```diff
@@ -8,62 +8,71 @@ export type LdOptions<T extends object> = {
   onError: (err: unknown) => void;
 } & RequestInit;
 
-const table = new Map()
+type TableMap<T> = Map<string, T|undefined>;
 
-function proxyfy<T extends object>(obj: T, options: LdOptions<T>): T {
+const cachedTable: TableMap<unknown> = new Map()
+
+function proxify<T extends object>(obj: T, options: LdOptions<T>): T {
   if (undefined === options.root) {
-    table.clear()
+    cachedTable.clear()
     options.root = obj
   }
 
   return new Proxy<T>(obj, {
-    get: (target, prop, receiver) => {
-      const value = Reflect.get(target, prop, receiver);
+    get: (target: T, prop: string, receiver: unknown) => {
+      const value: string extends keyof T ? T[string] : unknown = Reflect.get(target, prop, receiver);
 
       if (typeof value === 'object' && value !== null) {
-        return proxyfy<any>(value, options);
+        return proxify<any>(value, options);
       }
 
-      if (typeof value !== 'string') {
-        return value
+      if (typeof value !== 'string' && typeof value !== 'number' && typeof value !== 'symbol') {
```

Comment:
not sure why if the typeof value is not a string it is not a number nor a symbol right?

Reply:
Can probably remove them indeed, it would take too much to handle Symbols right now (since Symbols can be used as key, and therefore are "stringable"-ish)